atomicops.h

```diff
@@ -44,8 +44,18 @@
 #define AE_UNUSED(x) ((void)x)
 
 // AE_NO_TSAN/AE_TSAN_ANNOTATE_*
+// For GCC
+#if defined(__SANITIZE_THREAD__)
+#define _TSAN_IS_ENABLED 1
+#endif
+// For clang
 #if defined(__has_feature)
 #if __has_feature(thread_sanitizer)
+#define _TSAN_IS_ENABLED 1
+#endif
+#endif
+
+#if _TSAN_IS_ENABLED
```

Comment:
How does this compile? This introduces a new `#if` but _removes_ an `#endif` instead of adding a corresponding one.